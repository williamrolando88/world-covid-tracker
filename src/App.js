import React from 'react';
import { Route, Routes } from 'react-router-dom';
import HomePage from './components/HomePage';
import TodayDetails from './components/TodayDetails';

const App = () => {
  return (
    <>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/today/:country" element={<TodayDetails />} />
      </Routes>
    </>
  );
};

export default App;
