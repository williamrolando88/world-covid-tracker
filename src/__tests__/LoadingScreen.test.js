import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

const initialState = {
  covidData: {
    today: {
      status: 'pending',
    },
  },
};

const mockStore = configureStore();

describe('Check app snapshot', () => {
  test('should renderApp', () => {
    const app = renderer.create(
      <Provider store={mockStore(initialState)}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });
});
