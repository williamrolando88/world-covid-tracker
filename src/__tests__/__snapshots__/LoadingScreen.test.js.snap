// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check app snapshot should renderApp 1`] = `
<div
  className="fixed top-0 left-0 flex h-screen w-screen flex-col items-center justify-center  bg-[#302b63] text-stone-200"
>
  <img
    alt="Loading-icon"
    className="h-24 w-24 animate-spin invert"
    height="100"
    src="loading.svg"
    width="100"
  />
  <h2
    className="font-lato text-3xl font-bold"
  >
    Loading Data
  </h2>
</div>
`;
